webapp-django/crashstats/signature/views.py

```diff
@@ -424,3 +424,112 @@ def signature_graph_data(request, channel):
     )
 
     return data
+
+
+@waffle_switch('signature-report')
+def signature_summary(request):
+    '''Return a list of specific aggregations. '''
+    params = get_validated_params(request)
+    if isinstance(params, http.HttpResponseBadRequest):
+        # There was an error in the form, let's return it.
+        return params
```

Comment:
Hehe. It's the bane of our life! On the one hand we love short and few-line-PRs more than we love oxygen. On the other hand, PRs that repeat old common bad habits is our only chance to tackle technical debt. 

Just forget about it for now. The PR is already long. I often tackle refactoring/technical debt all-of-a-sudden. I.e. I have some downtime between important feature work and just attack it.